controllers/nstemplateset/cluster_resources.go

```diff
@@ -103,6 +103,7 @@ func (r *clusterResourcesManager) apply(ctx context.Context, nsTmplSet *toolchai
 		toolchainv1alpha1.TemplateRefLabelKey: tierTemplate.templateRef,
 		toolchainv1alpha1.TierLabelKey:        tierTemplate.tierName,
 		toolchainv1alpha1.ProviderLabelKey:    toolchainv1alpha1.ProviderLabelValue,
+		toolchainv1alpha1.OwnerLabelKey:       nsTmplSet.Namespace,
```

Comment:
should we use a different label ? `OwnerLabelKey` is usually used in a different way , by assigning it the value of the usersignup name . 

Maybe creating a new label like `NamespaceOwnerLabelKey` or `ClusterOwnerLabelKey` and passing the cluster name ?